tests/commands/try_repo_test.py

```diff
@@ -98,6 +101,37 @@ def test_try_repo_relative_path(cap_out, tempdir_factory):
         assert not try_repo(try_repo_opts(relative_repo, hook='bash_hook'))
 
 
+def test_try_repo_no_commits_no_git(cap_out, tempdir_factory):
+    repo = tempdir_factory.get()
+    with pytest.raises(CalledProcessError) as e:
+        try_repo(try_repo_opts(repo, hook='bash_hook'))
+    assert 'appears to have no commits' not in str(e.value)
+
+
+@pytest.mark.parametrize(
+    'has_commits', [
+        False,
+        True,
+    ],
+)
+def test_try_repo_no_commits(cap_out, tempdir_factory, has_commits):
+    repo = make_repo(
+        tempdir_factory,
+        'modified_file_returns_zero_repo',
+        commits=has_commits,
+    )
+
+    with cwd(git_dir(tempdir_factory)):
+        bare_repo = os.path.join(repo, '.git')
+        # previously crashed attempting modification changes
+        if not has_commits:
+            with pytest.raises(FatalError) as e:
+                try_repo(try_repo_opts(bare_repo, hook='bash_hook'))
+            assert 'appears to have no commits' in str(e.value)
+        else:
+            assert not try_repo(try_repo_opts(bare_repo, hook='bash_hook'))
```

Comment:
a few rules of thumb to follow:
- don't use booleans in parametrize (this is a special type of [boolean trap](https://youtu.be/CnRkXO_a5mI)
- don't mix success and failure in the same test
- don't [write logic in tests](https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html) -- your tests should be dead simple obvious what's going on and if you have to put `if` statements or loops in them then your tests likely aren't doing a good job at that

any of those -> these should be two disparate tests and not attempting to batch them together